compiler/ballerina-lang/src/main/java/io/ballerina/projects/repos/FileSystemCache.java

```diff
@@ -98,9 +98,19 @@ public void cacheBir(ModuleName moduleName, ByteArrayOutputStream birContent) {
     public Optional<Path> getPlatformSpecificLibrary(CompilerBackend compilerBackend, String libraryName,
                                                      boolean isOptimizedLibrary) {
         String libraryFileName = libraryName + compilerBackend.libraryFileExtension();
-        Path targetPlatformCacheDirPath = isOptimizedLibrary ? getOptimizedTargetPlatformCacheDirPath(compilerBackend) :
-                getTargetPlatformCacheDirPath(compilerBackend);
-        Path jarFilePath = targetPlatformCacheDirPath.resolve(libraryFileName);
+        Path optimizedJarFilePath = getOptimizedTargetPlatformCacheDirPath(compilerBackend).resolve(libraryFileName);
+        if (isOptimizedLibrary) {
+            return Optional.of(optimizedJarFilePath);
+        }
+
+        Path jarFilePath = getTargetPlatformCacheDirPath(compilerBackend).resolve(libraryFileName);
+        // getPlatformSpecificLibrary gets invoked from TestUtils class and there is no proper way to check whether a
+        // given module is `optimized`, `unoptimized` or `duplicated optimized`. Maybe we can lift the logic to
+        // JarResolver somehow.
+        // TODO Find a cleaner way to handle JarPaths for bal test with dead code elimination.
+        if (!Files.exists(jarFilePath) && Files.exists(optimizedJarFilePath)) {
+            return Optional.of(optimizedJarFilePath);
+        }
```

Comment:
We can simplify this as follows:
```java
if (Files.exists(jarFilePath)) {
     Optional.of(jarFilePath);
} else if (Files.exists(optimizedJarFilePath)) {
     Optional.of(optimizedJarFilePath);
}
return Optional.empty();
```

Reply:
Fixed in 4227209793366eef76a51a9240d5d92bf2033977.